compositor_pipeline/src/pipeline.rs

```diff
@@ -221,22 +278,22 @@ impl<Input: PipelineInput, Output: PipelineOutput> Pipeline<Input, Output> {
 
     pub fn with_outputs<F, R>(&self, f: F) -> R
     where
-        F: Fn(OutputIterator<'_, Output>) -> R,
+        F: Fn(OutputIterator<'_, PipelineOutput<Receiver>>) -> R,
```

Comment:
```suggestion
        F: Fn(OutputIterator<'_, Receiver>) -> R,
```

PipelineOutput should be a purely internal implementation, or if you want to expose both rtp and encode entities, maybe create sth on the fly that gives access to both.

Reply:
This is proving to be quite difficult in our current architecture. I think I would have to remove the generic type parameter from `OutputIterator` and replace it with something concrete, bc doing anything else with the current `OutputIterator` is quite difficult.

Reply:
do you think this would be fine?